client/src/components/CoinInput.tsx

```diff
@@ -73,9 +87,20 @@ export function useCoinInput({
     if (initialAmount != null) setAmount(initialAmount);
   }, [initialAmount]);
 
+  // TODO: consider real gas fee, replacing GAS_FEE variable.
+  // For now we need to keep 1 unit in the wallet(it's not spent) in order to complete "create pool" transaction.
+  function getSafeMaxBalance() {
+    const amount = coinBalance?.amount || BigInt(0);
+
+    return (amount > BigInt(0) && coin?.name === 'ETH') ?
```

Comment:
This should be on the screen also don't need to be inside the component

Reply:
I agree `GAS_FEE` can come from the screen, that makes sense. This calculation is just to set max balance value in input